### PR TITLE
imx7: use upstream u-boot for pico

### PR DESCRIPTION
**Pull request by niej**
use upstream u-boot for pico.

DCD file cannot be merge to mainline u-boot. Because there are two version pico board with 1GB/512MB DRAM. So u-boot guys insist u-boot SPL should be used to initialize DRAM, instead of DCD data. We have to carry it in recipe.

### CODE REVIEW

**Comment by niej** [comment]
http://jenkins.mbed-linux.arm.com/view/JunNie/job/jun-use-upstream-imx7-uboot/3/

test result is as below. Some WiFi test fail, but should not be related to this PR. RPI3 build failed, but should be jenkins issue.

http://e120856-lin.cambridge.arm.com/results.php?submitter=junnie01&image_version=custom&custom_version=jun-use-upstream-imx7-uboot&image_number=3&lava_query=

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend [hunk 1]
```diff
@@ -22,6 +26,10 @@ do_compile_append_imx7s-warp-mbl() {
 	oe_runmake u-boot-dtb.imx
 }
 
-do_deploy_append() {
+do_deploy_append_imx7s-warp-mbl() {
```
Comment (simonqhughes) [inline meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: why can't u-boot-fslc do_deploy_append() be the same for both imx7s and imx7d?

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2018-2019 Arm Limited and Contributors. All rights reserved.
```
Comment (simonqhughes) [inline meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout]: where did this file come from?

if this file is going to be called imx7d-pico-uboot-dtb.cfg, then the warp7 file should be called imx7s-warp-u-boot-dtb.cfg? I suggest you call the files ${MACHINE}-u-boot-dtb.cfg and make one do_deploy_depends() which is uses the suggested naming convention.

Comment (niej) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout]: This file is written by me, with values abstracted from u-boot SPL code and modified with referencing warp7 dcd file.

I can change the name. But dcd for warp7 is from ${B}, while dcd for pico is from ${WORKDIR}. How can we align the path, creating DCD_PATH_imx7s-warp-mbl DCD_PATH_imx7d-pico-mbl is acceptable?

File: meta-freescale-3rdparty-mbl/conf/machine/imx7d-pico-mbl.conf [hunk 3]
```diff
@@ -45,7 +45,7 @@ KERNEL_IMAGETYPE = "fitImage"
 SPL_BINARY=""
 UBOOT_EXTLINUX = "0"
 UBOOT_CONFIG = ""
-UBOOT_MACHINE = "pico-pi-imx7d_defconfig"
```
Comment (simonqhughes) [inline meta-freescale-3rdparty-mbl/conf/machine/imx7d-pico-mbl.conf]: The commit comment doesn't include the github PR explanation for why these changes are needed:

```
use upstream u-boot for pico.

DCD file cannot be merge to mainline u-boot. Because there are two version pico board with 1GB/512MB DRAM. So u-boot guys insist u-boot SPL should be used to initialize DRAM, instead of DCD data. We have to carry it in recipe.
```

Please can you update the commit comment to include more details? Thanks.

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout (hunk 2 above)
Comment (simonqhughes) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout]: >>This file is written by me, with values abstracted from u-boot SPL code 

The filel is therefore derived from u-boot spl code and that needs to be acknowledged in the licensing. See here for details:

https://confluence.arm.com/pages/viewpage.action?pageId=266055215

and this within the page: 

```
Based on: (OSS-FILE)
In open-source project: (OSS-URI)
 
Original file: (OSS-COPYRIGHT)
Modifications: Copyright (c) 2018 Arm Limited and Contributors. All rights reserved.
 
SPDX-License-Identifier: (OSS-LICENSE-SPDX)
```

>> reating DCD_PATH_imx7s-warp-mbl DCD_PATH_imx7d-pico-mbl is acceptable?

yes

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 # MBL_UBOOT_VERSION should be updated to match version pointed to by SRCREV
 MBL_UBOOT_VERSION = "2018.11-rc1"
 
 SRCREV = "c0c4ee5fce01ec0818c4f27ce029d9b16c8849ad"
+SRCREV_imx7d-pico-mbl = "c72a23701bd9b15dab35e0d43c468ab051739af3"
 
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
 SRC_URI = "git://git.linaro.org/landing-teams/working/mbl/u-boot.git;protocol=https;nobranch=1 "
+SRC_URI_imx7d-pico-mbl = "git://git.denx.de/u-boot.git"
```
Comment (texierp) [inline meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: should be gitlab.denx.de instead ? since it's now the official repo.

See https://lists.denx.de/pipermail/u-boot/2019-June/373065.html

Comment (bryanodonoghue) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: https://gitlab.denx.de/u-boot/u-boot.git

**Comment by niej** [comment]
lava test is as below. All passed wifi and avahi-discovery test fail. Not sure what avahi-discovery test, seems to be wireless related too.

http://e120856-lin.cambridge.arm.com/results.php?submitter=junnie01&image_version=custom&custom_version=jun-use-upstream-imx7-uboot&image_number=8&lava_query=

File: meta-freescale-3rdparty-mbl/conf/machine/imx7d-pico-mbl.conf (hunk 3 above)
Comment (niej) [reply meta-freescale-3rdparty-mbl/conf/machine/imx7d-pico-mbl.conf]: Already updated.

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend (hunk 1 above)
Comment (niej) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: explained in git comments.

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout (hunk 2 above)
Comment (niej) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc/imx7d-pico-u-boot-dtb.cfgout]: Done

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend (hunk 4 above)
Comment (niej) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: Already revised.

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend [hunk 5]
```diff
@@ -1,15 +1,19 @@
-# Copyright (c) 2018 Arm Limited and Contributors. All rights reserved.
+# Copyright (c) 2018-2019 Arm Limited and Contributors. All rights reserved.
 #
 # SPDX-License-Identifier: MIT
 
 # MBL_UBOOT_VERSION should be updated to match version pointed to by SRCREV
 MBL_UBOOT_VERSION = "2018.11-rc1"
 
 SRCREV = "c0c4ee5fce01ec0818c4f27ce029d9b16c8849ad"
+SRCREV_imx7d-pico-mbl = "c72a23701bd9b15dab35e0d43c468ab051739af3"
```
Comment (diego-sueiro) [inline meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: Why warp7 and pico7 can't have the same SHA?

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend (hunk 4 above)
Comment (diego-sueiro) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: @niej ,

Are you going to change to gitlab.denx.de/u-boot/u-boot ?

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend (hunk 5 above)
Comment (niej) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-fslc_%.bbappend]: There are several warp7 specific patches in u-boot, which are to be pushed to upstream.

File: meta-freescale-3rdparty-mbl/conf/machine/imx7d-pico-mbl.conf (hunk 3 above)
Comment (simonqhughes) [reply meta-freescale-3rdparty-mbl/conf/machine/imx7d-pico-mbl.conf]: thanks
